contracts/anchor/src/state.rs

```diff
@@ -0,0 +1,320 @@
+            assert!(self.max_edges > edge_count as u32, "Edge list is full");
+            save_edge(store, edge.chain_id, edge)?;
+            save_neighbor_roots(store, (edge.chain_id, 1), edge.root)?;
+            save_curr_neighbor_root_index(store, edge.chain_id, 1)?;
+            self.chain_id_list.push(edge.chain_id);
+        }
+
+        Ok(())
+    }
+
+    pub fn get_latest_neighbor_root(
+        &self,
+        chain_id: ChainId,
+        store: &dyn Storage,
+    ) -> Result<[u8; 32], ContractError> {
+        let neighbor_root_index = read_curr_neighbor_root_index(store, chain_id)
+            .map_err(|_| ContractError::ItemNotFound)?;
+
+        let latest_neighbor_root = read_neighbor_roots(store, (chain_id, neighbor_root_index))
+            .map_err(|_| ContractError::ItemNotFound)?;
+        Ok(latest_neighbor_root)
+    }
+
+    pub fn get_latest_neighbor_edges(&self, store: &dyn Storage) -> Vec<Edge> {
+        self.chain_id_list
+            .iter()
+            .map(|c_id| read_edge(store, *c_id).unwrap_or_default())
+            .collect()
+    }
+
+    pub fn get_neighbor_roots(&self, store: &dyn Storage) -> Vec<[u8; 32]> {
+        self.chain_id_list
+            .iter()
+            .map(|c_id| read_edge(store, *c_id).unwrap_or_default().root)
+            .collect()
+    }
+
+    pub fn is_known_neighbor_root(
+        &self,
+        chain_id: ChainId,
+        root: [u8; 32],
+        store: &dyn Storage,
+    ) -> bool {
+        if root == [0u8; 32] {
+            return false;
+        }
+
+        let mut i = read_curr_neighbor_root_index(store, chain_id).unwrap_or_default();
+        for _ in 0..ROOT_HISTORY_SIZE {
+            if let Ok(r) = read_neighbor_roots(store, (chain_id, i)) {
+                if r == root {
+                    return true;
+                }
+
+                if i == 0 {
+                    i = ROOT_HISTORY_SIZE - 1;
+                } else {
+                    i -= 1;
+                }
+            }
+        }
+
+        false
+    }
+
+    pub fn is_valid_neighbor_roots(&self, roots: &[[u8; 32]], store: &dyn Storage) -> bool {
+        assert!(
+            roots.len() == self.max_edges as usize - 1,
+            "Incorrect roots length"
+        );
+        for (i, edge) in self
+            .chain_id_list
+            .iter()
+            .map(|c_id| read_edge(store, *c_id).unwrap_or_default())
+            .enumerate()
+        {
+            if !self.is_known_neighbor_root(edge.chain_id, roots[i], store) {
+                return false;
+            }
+        }
+        true
+    }
+}
+
+// Anchor struct
+// TODO: Anchor should have an ERC20 attached
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Anchor {
+    pub chain_id: u64,
+    pub deposit_size: Uint256,
+    pub merkle_tree: MerkleTree,
+    pub linkable_tree: LinkableMerkleTree,
+    // used_nullifiers: Mapping<[u8; 32], bool>,
```

Comment:
Can we remove these commented out parts?

Reply:
Removed the comments as review.
https://github.com/webb-tools/protocol-cosmwasm/pull/15/commits/af5fc840e16bbe89fb2ee03062b8dca0f4e5acbd